strawberry/experimental/pydantic/object_type.py

```diff
@@ -76,6 +77,13 @@ def _build_dataclass_creation_fields(
             graphql_name = existing_field.graphql_name
         elif field.has_alias and use_pydantic_alias:
             graphql_name = field.alias
+
+        type_annotation: Optional[StrawberryAnnotation] = None
+        if not isinstance(field_type, StrawberryAnnotation):
+            type_annotation = StrawberryAnnotation(field_type)
+        else:
+            type_annotation = field_type
```

Comment:
should we wrap this into a staticmethod of StrawberryAnnotation? we do the same check below :)